plugins/proxy-sigv4-backend/pkg/config.d.ts

```diff
@@ -0,0 +1,33 @@
+export interface Config {
+  /**
+   * A list of forwarding-proxies. Each key is a route to match,
+   * below the prefix that the proxy plugin is mounted on. It must
+   * start with a '/'.
+   */
+  proxysigv4?: {
+    /**
+     * When `true`, configures an auth policy when using the new backend system to allow
+     * unauthenticated requests to the `proxy-sigv4-backend` router.
+     *
+     * Defaults to `true`.
+     */
+    allowUnauthenticatedRequests?: boolean;
+  } & {
+    [key: string]:
+      | string
+      | {
+          /**
+           * Target of the proxy. Url string to be parsed with the url module.
+           */
+          target: string;
+          /**
+           * Use credentials for this IAM Role ARN via STS:AssumeRole call when signing the request.
+           */
+          roleArn?: string;
+          /**
+           * Use this session name when performing the STS:AssumeRole call for roleArn. Default: 'tempAssumeRoleSession'.
+           */
+          roleSessionName?: string;
+        };
+  };
```

Comment:
I'm not sure we want to be adding the `pkg` directory here, but this does remind me that we should be updating the actual [`config.d.ts` file](https://github.com/segmentio/segment-backstage-plugins/blob/f497502c4306d49c2d600ed393d6b97ab8d40b16/plugins/proxy-sigv4-backend/config.d.ts#L18-L31) for these new options, sorry for the late catch!